### PR TITLE
US4 and Significant Refactor of data exchange between activities

### PR DESCRIPTION
**Pull request by llunn**
4 of 35 Instrumented tests are failing.  The workout tests need more than a couple lines of changes to correct them.

### CODE REVIEW

File: app/src/test/java/ca/dal/cs/athletemonitor/athletemonitor/TeamTest.java [hunk 1]
```diff
@@ -10,33 +10,42 @@
 /**
  * Unit tests for the Team class
  */
-
 public class TeamTest {
```
Comment (nickburris) [inline app/src/test/java/ca/dal/cs/athletemonitor/athletemonitor/TeamTest.java]: Add @RunWith(AndroidJUnit4.class), clearly android studio notices the tests anyway, but just to be safe

File: app/src/androidTest/java/ca/dal/cs/athletemonitor/athletemonitor/CreateTeamActivityTest.java [hunk 2]
```diff
@@ -72,9 +104,35 @@ public void testSubmitButtonEnabled() throws Exception {
...
+
+    /**
+     * Tests that creating a team when clicking create team button
+     *
+     * @throws Exception General exception
+     */
+    @Test
+    public void testOnCreateTeamButtonClick() throws Exception {
+        // add a team to the user and update the user informatoin in firebase
+        testUser.addUserTeam(new Team("testteam", "testmotto", testUser.getUsername()));
```
[the remark this answers is not in the document]
Comment (llunn) [reply app/src/androidTest/java/ca/dal/cs/athletemonitor/athletemonitor/CreateTeamActivityTest.java]: Leftover from iteration 1 i think.  Good catch

File: app/src/test/java/ca/dal/cs/athletemonitor/athletemonitor/TeamTest.java (hunk 1 above)
Comment (llunn) [reply app/src/test/java/ca/dal/cs/athletemonitor/athletemonitor/TeamTest.java]: I think this is not possible for unit test

Comment (nickburris) [reply app/src/test/java/ca/dal/cs/athletemonitor/athletemonitor/TeamTest.java]: My bad, thought this was a espresso test, in that case I believe it should be `@RunWith(JUnit4.class)`

Comment (llunn) [reply app/src/test/java/ca/dal/cs/athletemonitor/athletemonitor/TeamTest.java]: Ack, i added it to all of the unit test classes

**Pull request merged**